cypress/e2e/MovieListPage.cy.ts

```diff
@@ -0,0 +1,54 @@
+describe('Movie List Page', () => {
+    beforeEach(() => {
+        cy.visit('http://localhost:3000')
+    })
+
+    it('successfully loads', () => {
+        cy.url().should('include', '/')
+    })
+
+    it('genre select works correctly', () => {
+        // Select the genre
+        cy.get('.genre-select').contains('Comedy').click()
+
+        // Verify the genre is selected
+        cy.get('.genre-select')
+            .find('.selected-button')
+            .should('contain.text', 'Comedy')
+    })
+
+    it('sort control works correctly', () => {
+        cy.get('.movie-list-page-navbar')
+            .find('select')
+            .select('release_date')
+            .should('have.value', 'release_date')
+    })
+
+    it('displays movies', () => {
+        cy.get('.movie-list').children().should('have.length.greaterThan', 0)
+    })
+
+    it('movie counter displays correct number of movies', () => {
+        cy.get('.movie-counter')
+            .invoke('text')
+            .then(text => {
+                const movieCount = parseInt(text.split(' ')[0])
+                cy.get('.movie-list')
+                    .children()
+                    .should('have.length', movieCount)
```

Comment:
This test does not work properly because it does not wait for movies to load. As a result, `cy.get('.movie-list').children()` is `undefined`. Please wait for movies to load before making such assertions.